testing/integration/upgrade_test.go

```diff
@@ -208,18 +211,18 @@ func TestStandaloneUpgrade(t *testing.T) {
 
 			parsedUpgradeVersion, err := version.ParseVersion(define.Version())
 			require.NoErrorf(t, err, "define.Version() %q cannot be parsed as agent version", define.Version())
-			testStandaloneUpgrade(ctx, t, agentFixture, parsedVersion, parsedUpgradeVersion, "", true)
+			testStandaloneUpgrade(ctx, t, agentFixture, parsedVersion, parsedUpgradeVersion, "", true, true, false, "")
 		})
 	}
 }
 
-func TestStandaloneUpgradeToSpecificSnapshotBuild(t *testing.T) {
+func TestStandaloneUpgradeToSpecificSnapshotBuildWithGPGFallback(t *testing.T) {
```

Comment:
I still don't understand why this test has to use a different build of the same snapshot version for the upgrade...
Can't we test the same functionality by downgrading from the locally built package to a previous (distinct) version ? Would that change the nature of the test?

If we can choose a different version you can simplify the test to a `testStandaloneUpgrade` call + a few statements to manipulate the PGP key as it's already there... You have no need to the commit sha workaround or checking the commit sha after upgrade...

Reply:
updated to simpler version